### PR TITLE
Added ability to restore position with the _exit_point

### PR DESCRIPTION
**Pull request by Frix-x**
Hey, here is mainly an aesthetic PR to simplify some macros.

First I added the option to restore the nozzle position using the _exit_point macro at a set speed.
Secondly, this also enable us to simplify the PROBE_ACCURACY and PROBE_CALIBRATE custom overrides by removing/replacing the multiples SAVE_GCODE_STATE and RESTORE_GCODE_STATE calls.

### CODE REVIEW

File: Klipper_macro/klicky-probe.cfg [hunk 1]
```diff
@@ -380,38 +381,35 @@ gcode:
         { action_raise_error("Must Home X, Y and Z Axis First!") }
     {% endif %}
 
-    _entry_point function=PROBE_CALIBRATE
-
     # Go to Z safe distance before saving location in order to
     # avoid crashing the probe on the bed when coming back
     {% if (printer.toolhead.position.z < Hzh) %}
+        G90
```
Comment (jlas1) [inline Klipper_macro/klicky-probe.cfg:387]: By forcing a G90 without doing a save first, you lose the previous state, that is not desired.

File: Klipper_macro/klicky-probe.cfg [hunk 2]
```diff
@@ -428,39 +426,35 @@ gcode:
         { action_raise_error("Must Home X, Y and Z Axis First!") }
     {% endif %}
 
-    _entry_point function=PROBE_ACCURACY
-
     # Go to Z safe distance before saving location in order to
     # avoid crashing the probe on the bed when coming back
     {% if (printer.toolhead.position.z < Hzh) %}
+        G90
```
Comment (jlas1) [inline Klipper_macro/klicky-probe.cfg:432]: By forcing a G90 without doing a save first, you lose the previous state, that is not desired.

File: Klipper_macro/klicky-probe.cfg [hunk 3]
```diff
@@ -86,9 +86,10 @@ gcode:
 gcode:
     {% set function  = 'pre_' ~ params.FUNCTION  %}
     {% set move  = params.MOVE|default(0)  %}
+    {% set speed  = params.MOVE_SPEED|default(0) %}
     # mandatory to save the new safe position
     M400
-    RESTORE_GCODE_STATE NAME={function} MOVE={move}
+    RESTORE_GCODE_STATE NAME={function} MOVE={move} MOVE_SPEED={speed}
```
Comment (jlas1) [inline Klipper_macro/klicky-probe.cfg:92]: nice idea

**Review by jlas1 — CHANGES_REQUESTED** [review]
Please check the comments, the simplification goes against the objective of having the code always reverting to the state that it was before being called

File: Klipper_macro/klicky-probe.cfg (hunk 1 above)
Comment (Frix-x) [reply Klipper_macro/klicky-probe.cfg:387]: Oh yes, you're right, I did not think about it !

So even if the main modification is cool (add move/speed control to _exit_point), the simplification of PROBE_CALIBRATE and PROBE_ACCURACY is not really possible due to that G90. I'll try to see how to do it the best without this regression.

File: Klipper_macro/klicky-probe.cfg (hunk 3 above)
Comment (Frix-x) [reply Klipper_macro/klicky-probe.cfg:92]: This is the main subject of the PR to remove the need of all the SAVE/RESTORE_GCODE_STATE and use only the _entry_point and _exit_point :)

**Comment by jlas1** [comment]
Can you update the files on the PR? You can update them on your branch that it will be updated here.

**Comment by jlas1** [comment]
I'm gonna close this PR as the macros have changed a lot.
I have included the speed function on the _exit_point macro, thanks for that.

**Comment by Frix-x** [comment]
Sorry I was on vacation until today without proper internet access.
No problem, speed in the _exit_point macro was the main point of the PR :)

**Pull request closed without merging**